test/Microsoft.Azure.WebJobs.Host.EndToEndTests/InvocationFilterEndToEndTests.cs

```diff
@@ -0,0 +1,460 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Xunit;
+using Microsoft.Azure.WebJobs.Host.Timers;
+using Microsoft.Azure.WebJobs.Host.Loggers;
+
+namespace Microsoft.Azure.WebJobs.Host.EndToEndTests
+{
+    public class InvocationFilterEndToEndTests
+    {
+        private readonly TestLoggerProvider _loggerProvider = new TestLoggerProvider();
+        
+        [Fact]
+        public async Task TestFilterOrdering()
+        {
+            // Setup the test in memory
+            var activator = new FakeActivator();
+            var loggerFactory = new LoggerFactory();
+            var logger = new MyLogger();
+            activator.Add(new FilterOrderTestClass());
+            var host = TestHelpers.NewJobHost<FilterOrderTestClass>(activator, logger);
+            loggerFactory.AddProvider(_loggerProvider);
+            host.Configuration.LoggerFactory = loggerFactory;
+            host.Configuration.TypeLocator = new FakeTypeLocator(typeof(FilterOrderTestClass), typeof(FilterOrderTestClass.SampleClassForFilterOrderTest));
+            var method = typeof(FilterOrderTestClass).GetMethod("Main", BindingFlags.Public | BindingFlags.Instance);
```

Comment:
> var method = typeof(FilterOrderTestClass).GetMethod("Main", BindingFlags.Public | BindingFlags.Instance); [](start = 11, length = 106)

You should be able to use CAllAsync(string) and pass in the method name directly. No MethodInfo needed!